src/chapters/unit-testing/why-test.rst

```diff
@@ -0,0 +1,56 @@
+Why Test Your Code?
+===================
+
+Testing your code is part of the development process. Have you ever written a program and then walked away thinking, *I'm done!*, without having run it? Of course not. You are already used to testing and debugging programs as you write them. In fact, we devoted an entire chapter to :ref:`debugging <errors-and-debugging>` early in the course. 
+
+Your development process probably looks something like this:
+
+#. Write code
+#. Run program
+#. Notice error and investigate
+#. Repeat these steps until there are no more errors
+
+.. index:: ! unit testing
+
+But there's a better way to test your code, using *automated* tests. There are many types of automated tests. This chapter focuses on **unit testing**, which tests the smallest components (or *units*) of code. These are typically individual functions.
+
+Before we dive into the *how* of unit testing, let's discuss the *why*.
+
+Know Your Code *Really* Works
+-----------------------------
+
+Manual testing can lead you to a complete, error-free program. But unit testing does even more than that.
```

Comment:
It's not a guarantee. I agree with "can lead"

Reply:
I see your point, in this case. Applied to TDD, however, "leads" would be the better option, since it reinforces the conclusion we want the students to make.